apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/assessmentQuestionTable.tsx

```diff
@@ -0,0 +1,791 @@
+    }));
+  }, [aiGradingMode, setColumnVisibility]);
+
+  const table = useReactTable({
+    data: instanceQuestions,
+    columns,
+    columnResizeMode: 'onChange',
+    getRowId: (row) => row.id,
+    state: {
+      sorting,
+      columnFilters,
+      globalFilter,
+      columnSizing,
+      columnVisibility,
+      columnPinning,
+      rowSelection,
+    },
+    initialState: {
+      columnPinning: DEFAULT_PINNING,
+      columnVisibility: defaultColumnVisibility,
+    },
+    onSortingChange: setSorting,
+    onGlobalFilterChange: setGlobalFilter,
+    onColumnSizingChange: setColumnSizing,
+    onColumnVisibilityChange: setColumnVisibility,
+    onColumnPinningChange: setColumnPinning,
+    onRowSelectionChange: setRowSelection,
+    enableRowSelection: true,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    defaultColumn: {
+      minSize: 100,
+      size: 150,
+      maxSize: 500,
+      enableSorting: true,
+      enableHiding: true,
+      enablePinning: true,
+    },
+  });
+
+  // Ref for the checkbox element to set indeterminate state
+  const studentInfoCheckboxRef = useRef<HTMLInputElement>(null);
+
+  // Determine student info checkbox state based on column visibility
+  const studentInfoCheckboxState = useMemo(() => {
+    const visibility = columnVisibility;
+    const nameVisible = visibility.user_or_group_name;
+    const uidVisible = visibility.uid;
+
+    if (nameVisible && uidVisible) return 'checked';
+    if (nameVisible || uidVisible) return 'indeterminate';
+    return 'unchecked';
+  }, [columnVisibility]);
+
+  // Set indeterminate state on the checkbox element
+  useEffect(() => {
+    if (studentInfoCheckboxRef.current) {
+      studentInfoCheckboxRef.current.indeterminate = studentInfoCheckboxState === 'indeterminate';
+    }
+  }, [studentInfoCheckboxState]);
+
+  // Handle student info checkbox click - toggles between checked (both visible) and unchecked (both hidden)
+  const handleStudentInfoCheckboxClick = () => {
+    const currentState = studentInfoCheckboxState;
+    const currentVisibility = table.getState().columnVisibility;
+
+    let newVisibility: VisibilityState;
+    if (currentState === 'checked') {
+      // Checked -> Unchecked (hide both)
+      newVisibility = {
+        ...currentVisibility,
+        user_or_group_name: false,
+        uid: false,
+      };
+    } else {
+      // Unchecked or Indeterminate -> Checked (show both)
+      newVisibility = {
+        ...currentVisibility,
+        user_or_group_name: true,
+        uid: true,
+      };
+    }
+
+    void setColumnVisibility(newVisibility);
+  };
+
+  const selectedRows = table.getSelectedRowModel().rows;
+  const selectedIds = selectedRows.map((row) => row.original.id);
+
+  // Set up handlers for edit points popovers
+  // FIXME: Clean up this useEffect once other pages handling point changes are also in React.
+  useEffect(() => {
```

Comment:
This is super gross -- we should get out of this escape hatch ASAP (only possible after migration of other edit points popovers on other pages)